### PR TITLE
example-deployment:Add fluentd and MinIO to kubernetes cluster

### PR DESCRIPTION
**Pull request by matiasbavera**
## What's new
Related to #281

* Updated Readme with installation instructions of Fluentd and Minio
* Created a Fluentd configmap file for Fluentd and a deploy file.
* Created a MinIO deploy file with an ingress on `example.com/minio`
* Fluentd collects all logs and sends those logs to MinIO where are stored in chunks of 5mb. 

image

## Self-checks

- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: example-deployment/README.md [hunk 1]
```diff
@@ -246,6 +246,60 @@ deploy it
 .bin/minikube kubectl -- apply -f k8s/dashboard.yaml
 ```
 
+## [MinIO](https://github.com/minio/minio)
```
Comment (koonpeng) [inline example-deployment/README.md:249]: add minio to the list of apps deployed at the top?

File: example-deployment/README.md [hunk 2]
```diff
@@ -246,6 +246,60 @@ deploy it
...
+
+``` bash
+# connected to internet
+docker pull minio/minio:RELEASE.2021-03-10T05-11-33Z
+
+# connected to airgap network
+docker save minio/minio:RELEASE.2021-03-10T05-11-33Z | bash -c 'eval $(.bin/minikube docker-env) && docker load'
+
+.bin/minikube kubectl -- apply -f k8s/minio.yaml
+```
```
Comment (koonpeng) [inline example-deployment/README.md:260]: The docker instructions are not normally needed, when you deploy to k8s, it should auto fetch the images.

For airgapped deployment, I think it's better to direction people to the airgapped section instead of assuming (and providing) only the airgapped instructions.

File: example-deployment/README.md [hunk 3]
```diff
@@ -246,6 +246,60 @@ deploy it
...
+
+Official docker image of [fluentd](https://hub.docker.com/r/fluent/fluentd/). <br/> We adjusted `fluentd` to send logs to a local storage server called MinIO.
+
+``` bash
+# connected to internet
+docker pull fluent/fluentd-kubernetes-daemonset:v1.12.2-debian-s3-1.0
+
+# connected to airgap network
+docker save fluent/fluentd-kubernetes-daemonset:v1.12.2-debian-s3-1.0 | bash -c 'eval $(.bin/minikube docker-env) && docker load'
+```
```
Comment (koonpeng) [inline example-deployment/README.md:279]: See above comments.

File: example-deployment/README.md [hunk 4]
```diff
@@ -246,6 +246,60 @@ deploy it
...
+
+### Fluentd Configmap
+
+We have 4 files in our `fluentd-configmap.yaml` :
+* `fluent.conf`: Our main config which includes all configurations we want to run.
+* `pods-fluent.conf`: `tail` config that sources all pod logs on the `kubernetes` host in the cluster.
+* `minio-fluent.conf`: `match` config to capture all logs and send them to MinIO. Every chunck of logs should have 5mb. </br>
```
Comment (koonpeng) [inline example-deployment/README.md:286]: `</br>` formatting error?

File: example-deployment/k8s/fluentd-configmap.yaml [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+metadata:
+  name: fluentd-config
+data:
+  fluent.conf: |-
+    ################################################################
+    # This source gets all logs from local docker host
+    @include pods-fluent.conf
+    @include minio-fluent-dev.conf
+    # Use this configuration for production
+    # @include minio-fluent.conf
```
Comment (koonpeng) [inline example-deployment/k8s/fluentd-configmap.yaml:12]: default to production config?

File: example-deployment/k8s/fluentd-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      s3_endpoint http://minio-service:9000/  # The endpoint URL
+      # s3_region us-east-1           # See the region settings of your Minio server
+      path logs/                    # This prefix is added to each file
+      force_path_style true         # This prevents AWS SDK from breaking endpoint URL
+      time_slice_format %Y%m%d%H%M  # This timestamp is added to each file name
+      <buffer time>
+        @type file
+        path /var/log/fluent/s3
+        timekey 1m                # Flush the accumulated chunks every hour
+        timekey_wait 1s             # Wait for 60 seconds before flushing
```
Comment (koonpeng) [inline example-deployment/k8s/fluentd-configmap.yaml:44]: the comments are wrong

File: example-deployment/k8s/fluentd.yaml [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        version: v1
+    spec:
+      serviceAccount: fluentd
+      serviceAccountName: fluentd
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: fluentd
+        imagePullPolicy: "Always"
```
Comment (koonpeng) [inline example-deployment/k8s/fluentd.yaml:63]: this wouldn't work in airgapped network?

File: example-deployment/k8s/minio.yaml [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+    # The volume is mounted as read-write by a single node
+    - ReadWriteOnce
+  resources:
+    # This is the request for storage. Should be available in the cluster.
+    requests:
+      storage: 10Gi
+---
+
+apiVersion: v1
+kind: Service
```
Comment (koonpeng) [inline example-deployment/k8s/minio.yaml:84]: put service before deployment? https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables

**Review by koonpeng — APPROVED** [review]

File: example-deployment/README.md (hunk 2 above)
Comment (matiasbavera) [reply example-deployment/README.md:260]: Yeah, you're right, I'll do that!. Thanks!

File: example-deployment/README.md (hunk 1 above)
Comment (matiasbavera) [reply example-deployment/README.md:249]: Done! Thanks!

File: example-deployment/README.md (hunk 4 above)
Comment (matiasbavera) [reply example-deployment/README.md:286]: Yeah, it was a leftover. Thanks!

File: example-deployment/k8s/minio.yaml (hunk 8 above)
Comment (matiasbavera) [reply example-deployment/k8s/minio.yaml:84]: Oh, I didn't know that! thank you!

File: example-deployment/k8s/fluentd.yaml (hunk 7 above)
Comment (matiasbavera) [reply example-deployment/k8s/fluentd.yaml:63]: Yeah, you're right, I removed the ImagePullPolicy
